DOCS-1198: Improve examples and advanced MR (#2079)
viamrobotics/docs: docs/modular-resources/advanced/_index.md

    "components",
    "services",
  ]
description: "Some usage may require you to extend the standardized API definitions or deploy custom components using a server on a remote part"
aliases:
  - "/program/extend/"
---

> Some usage may require you to extend the standardized API definitions or deploy custom components using a server on a {{< glossary_tooltip term_id="remote" text="remote part" >}}.

Reviewer: ```suggestion
Some use cases may require you to define a new API, or deploy a custom component using a server on a {{< glossary_tooltip term_id="remote" text="remote part" >}}.
```
see above

> ## Custom Components as Remotes

Reviewer: Separately, I'd vote for moving this section to the end because I think this may be the more rare edge case versus new API subtypes

> [Modular resources](/modular-resources/) running on the board directly connected to your components are the preferred way of managing and controlling custom components.

Reviewer 2: ```suggestion
[Modular resources](/modular-resources/) running on the [board](/components/board/) directly connected to your components are the preferred way of managing and controlling custom components.
```

> [Modular resources](/modular-resources/) running on the [board](/components/board/) directly connected to your components are the preferred way of managing and controlling custom components.

Reviewer: ```suggestion
Running [modular resources](/modular-resources/) on the [board](/components/board/) directly connected to your components is the preferred way of managing and controlling custom components.
```

If you want to use most of an existing API but need just a few other functions, try using the [`DoCommand`](/program/apis/#docommand) endpoint and [extra parameters](/program/use-extra-params/) to add custom functionality to an existing subtype.

If your resource does not fit into any of the existing {{< glossary_tooltip term_id="component" text="component" >}} or {{< glossary_tooltip term_id="service" text="service" >}} {{< glossary_tooltip term_id="subtype" text="subtypes" >}} or you want to define different methods for the API, you can [define a new resource subtype and an API for that subtype](/modular-resources/advanced/create-subtype/).

> ## Custom Components as Remotes

Author: ```suggestion
## Custom components as remotes
```

If you want to use most of an existing API but need just a few other functions, try using the [`DoCommand`](/program/apis/#docommand) endpoint and [extra parameters](/program/use-extra-params/) to add custom functionality to an existing subtype.

> If your resource does not fit into any of the existing {{< glossary_tooltip term_id="component" text="component" >}} or {{< glossary_tooltip term_id="service" text="service" >}} {{< glossary_tooltip term_id="subtype" text="subtypes" >}} or you want to define different methods for the API, you can [define a new resource _subtype_ and an API for that subtype](/modular-resources/advanced/create-subtype/).

Reviewer: ```suggestion
If your resource does not fit into any of the existing {{< glossary_tooltip term_id="component" text="component" >}} or {{< glossary_tooltip term_id="service" text="service" >}} {{< glossary_tooltip term_id="subtype" text="subtypes" >}} or you want to define different methods for the API, you can [define a new resource subtype and an API for that subtype](/modular-resources/advanced/create-subtype/).
```
[nit] This just got used several times before so I don't think italics make sense after the first usage
